src/background/target-page-controller.ts

```diff
@@ -57,15 +58,19 @@ export class TargetPageController {
     private addKnownTabId = async (tabId: number) => {
         if (!this.knownTabIds.includes(tabId)) {
             this.knownTabIds.push(tabId);
-            await this.idbInstance.setItem(IndexedDBDataKeys.knownTabIds, this.knownTabIds);
+            if (this.persistStoreData) {
+                await this.idbInstance.setItem(IndexedDBDataKeys.knownTabIds, this.knownTabIds);
+            }
         }
     };
 
     private removeKnownTabId = async (tabId: number, context: TabContext) => {
         if (this.knownTabIds.includes(tabId)) {
             this.knownTabIds.splice(this.knownTabIds.indexOf(tabId, 0), 1);
-            context.teardown();
-            await this.idbInstance.setItem(IndexedDBDataKeys.knownTabIds, this.knownTabIds);
+            if (this.persistStoreData) {
+                context.teardown();
```

Comment:
Should this be outside the `if`? I think it should probably be the context's responsibility to decide whether its `teardown` work involves persisting store data or not. (I think in practice it just defers to stores' `teardown`s, which should already handle deciding whether to skip idb updates or not appropriately with the `PersistentStore.teardown()` change you've already made